### PR TITLE
[idea] #399 Fixed exceptions when plugin is disconnected

### LINKED ISSUE #399: Got if I disconnect while running the plugin.
Not a JSON Object: "<html>Elastic"
java.lang.IllegalStateException: Not a JSON Object: "<html>Elastic"
	at com.google.gson.JsonElement.getAsJsonObject(JsonElement.java:90)
	at com.imaginea.kodebeagle.util.ESUtils.getJsonElements(ESUtils.java:131)
	at com.imaginea.kodebeagle.util.ESUtils.fetchContentsAndUpdateMap(ESUtils.java:79)
	at com.imaginea.kodebeagle.util.ESUtils.getContentsForFile(ESUtils.java:97)
	at com.imaginea.kodebeagle.tasks.FetchFileContentTask.run(FetchFileContentTask.java:55)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:563)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:152)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:452)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:402)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:137)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl$1.run(ProgressManagerImpl.java:126)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:400)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
	at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:56)

java.lang.IllegalStateException: Not a JSON Object: "<html>Elastic"
	at com.google.gson.JsonElement.getAsJsonObject(JsonElement.java:90)
	at com.imaginea.kodebeagle.util.ESUtils.getJsonElements(ESUtils.java:131)
	at com.imaginea.kodebeagle.util.ESUtils.fetchContentsAndUpdateMap(ESUtils.java:79)
	at com.imaginea.kodebeagle.util.ESUtils.getContentsForFile(ESUtils.java:97)
	at com.imaginea.kodebeagle.tasks.FetchFileContentTask.run(FetchFileContentTask.java:55)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:563)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:152)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:452)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:402)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:137)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl$1.run(ProgressManagerImpl.java:126)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:400)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
	at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:56)

java.lang.IllegalStateException: Not a JSON Object: "<html>Elastic"
	at com.google.gson.JsonElement.getAsJsonObject(JsonElement.java:90)
	at com.imaginea.kodebeagle.util.ESUtils.getJsonElements(ESUtils.java:131)
	at com.imaginea.kodebeagle.util.ESUtils.fetchContentsAndUpdateMap(ESUtils.java:79)
	at com.imaginea.kodebeagle.util.ESUtils.getContentsForFile(ESUtils.java:97)
	at com.imaginea.kodebeagle.tasks.FetchFileContentTask.run(FetchFileContentTask.java:55)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:563)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:152)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:452)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:402)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:137)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl$1.run(ProgressManagerImpl.java:126)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:400)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
	at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:56)

### PR DESCRIPTION
**Pull request by keerath**
fixes #399

### CODE REVIEW

**Comment by keerath** [comment]
@ScrapCodes PTAL..!!

File: plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/util/ESUtils.java [hunk 1]
```diff
@@ -104,22 +106,23 @@ public final String getContentsForFile(final String fileName) {
...
-            String fileName = sourceObject.getAsJsonPrimitive(FILE).getAsString();
-            //Extracting repoIds for future use
-            int repoId = sourceObject.getAsJsonPrimitive(REPO_ID).getAsInt();
-            String project = getProjectName(fileName);
-            if (!windowObjects.getRepoNameIdMap().containsKey(project)) {
-                windowObjects.getRepoNameIdMap().put(project, repoId);
+        if (hitsObject != null) {
+            JsonArray hitsArray = getJsonHitsArray(hitsObject);
+            resultCount = hitsArray.size();
+            totalHitsCount = getTotalHits(hitsObject);
```
Comment (ScrapCodes) [inline plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/util/ESUtils.java]: This can also be null.

**Comment by keerath** [comment]
@ScrapCodes is this fine .. ??

File: plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/util/ESUtils.java [hunk 2]
```diff
@@ -76,8 +76,9 @@ public final void fetchContentsAndUpdateMap(final List<String> fileNames) {
         String esFileResultJson;
         esFileResultJson = getESResultJson(esFileQueryJson,
                 windowObjects.getEsURL() + SOURCEFILE_SEARCH);
-        JsonArray hitsArray = getJsonHitsArray(getJsonElements(esFileResultJson));
-
+        JsonObject jsonElements = getJsonElements(esFileResultJson);
+        assert jsonElements != null;
```
Comment (ScrapCodes) [inline plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/util/ESUtils.java]: These can enabled or disabled. See http://docs.oracle.com/cd/E19683-01/806-7930/6jgp65ikq/index.html

Comment (ScrapCodes) [inline plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/util/ESUtils.java]: Please use Preconditions.check* from google guava

**Pull request closed without merging**